### PR TITLE
[Gluten-core] Support scan-only execution

### PR DESCRIPTION
**Pull request by rui-mo**
Support scan-only execution. When enabled, only scan and the filter after scan will be offloaded to native.

### CODE REVIEW

**Comment by jinchengchenghh** [comment]
This config is just for our local test and user special requirement. Can you keep it in your repo, not merge to gluten? Because it will make gluten logic complicate.

**Comment by FelixYBW** [comment]
> This config is just for our local test and user special requirement. Can you keep it in your repo, not merge to gluten? Because it will make gluten logic complicate.

Let's add this quick solution for now, It can help to debug the parquet reader issue quickly. If we don't need it in future, we can revert the PR.

File: gluten-core/src/main/scala/io/glutenproject/extension/columnar/TransformHintRule.scala [hunk 1]
```diff
@@ -319,22 +321,22 @@ case class AddTransformHintRule() extends Rule[SparkPlan] {
           //   and need some tests before being enabled.
           TransformHints.tagNotTransformable(plan)
         case plan: ProjectExec =>
-          if (!enableColumnarProject) {
+          if (!enableColumnarProject || scanOnly) {
```
Comment (FelixYBW) [inline gluten-core/src/main/scala/io/glutenproject/extension/columnar/TransformHintRule.scala:324]: Can we move scanOnly when we set "enableColumnarProject"? Which means the scan only option will overwrite all other operator's enablexxx config.

File: shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala [hunk 2]
```diff
@@ -144,6 +148,10 @@ class GlutenConfig(conf: SQLConf) extends Logging {
   def enableLogicalJoinOptimize: Boolean =
     conf.getConfString("spark.gluten.sql.columnar.logicalJoinOptimizeEnable", "false").toBoolean
 
+  // When enabled, only scan and the filter after scan will be offloaded to native.
+  def enableScanOnly: Boolean =
```
Comment (FelixYBW) [inline shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala:152]: Can you add the config to readme as well? highlight the config will overwrite all other operators' enabling. Just offload parquet scan and filter pushdown.

File: gluten-core/src/main/scala/io/glutenproject/extension/columnar/TransformHintRule.scala (hunk 1 above)
Comment (rui-mo) [reply gluten-core/src/main/scala/io/glutenproject/extension/columnar/TransformHintRule.scala:324]: Does that mean putting `scanOnly` before `!enableColumnarProject`?

Comment (FelixYBW) [reply gluten-core/src/main/scala/io/glutenproject/extension/columnar/TransformHintRule.scala:324]: No, use something like 
enableColumnarProject = !scanonly && config.get(xxx)

Comment (rui-mo) [reply gluten-core/src/main/scala/io/glutenproject/extension/columnar/TransformHintRule.scala]: Fixed.

File: shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala (hunk 2 above)
Comment (rui-mo) [reply shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala:148]: Updated doc.

**Comment by Yohahaha** [comment]
> This config is just for our local test and user special requirement. Can you keep it in your repo, not merge to gluten? Because it will make gluten logic complicate.

+1.

**Comment by rui-mo** [comment]
Merging this PR as a quick solution.

**Pull request merged**